internal/git/git.go

```diff
@@ -183,40 +185,49 @@ func (c Client) FindExtinctions(projKey string, flags []string, matcher search.M
 		if err != nil {
 			return nil, err
 		}
-		patch, err := changes.Patch()
+		patch, err := changes.PatchContext(context.Background())
 		if err != nil {
 			return nil, err
 		}
-		patchLines := strings.Split(patch.String(), "\n")
-		nextFlags := make([]string, 0, len(flags))
-		for _, flag := range flags {
-			removalCount := 0
-			for _, patchLine := range patchLines {
-				delta := 0
-				// Is a change line and not a metadata line
-				if strings.HasPrefix(patchLine, "-") && !strings.HasPrefix(patchLine, "---") {
-					delta = 1
-				} else if strings.HasPrefix(patchLine, "+") && !strings.HasPrefix(patchLine, "+++") {
-					delta = -1
-				}
-				if delta != 0 && matcher.MatchElement(patchLine, flag) {
-					removalCount += delta
+		for _, filePatch := range patch.FilePatches() {
+			fromFile, toFile := filePatch.Files()
+			if project.Dir != "" && (toFile == nil || !strings.HasPrefix(toFile.Path(), project.Dir)) {
```

Comment:
Tested this logic with local changes. Need to add more test coverage, also would like to talk over the implementation.

Reply:
We added validation logic in a separate PR building on this one that the `project.Dir` has to be relative. Given that and how go-git returns paths as relative to the root of the repo this and the conditional inside it are checking if the path does not start the `projects` `dir`. If not then continue onto next patch